ui-trackoccupancydiagram/src/components/helpers/drawElements/drawTrack.ts

```diff
@@ -0,0 +1,81 @@
+import { sum } from 'lodash';
+
+import { TRACK_HEIGHT_CONTAINER, COLORS, TICKS_PATTERN } from '../../consts';
+import { getTickPattern } from '../../utils';
+
+const { WHITE_50, GREY_20, RAIL_TICK } = COLORS;
+
+const drawRails = ({
+  xStart,
+  yStart,
+  width,
+  stroke = GREY_20,
+  ctx,
+}: {
+  xStart: number;
+  yStart: number;
+  width: number;
+  stroke?: string;
+  ctx: CanvasRenderingContext2D;
+}) => {
+  ctx.clearRect(xStart, yStart, width, 9);
+  ctx.beginPath();
+  ctx.fillStyle = WHITE_50;
+  ctx.strokeStyle = stroke;
+  ctx.lineWidth = 1;
+  ctx.rect(xStart, yStart, width, 8);
+  ctx.fill();
+  ctx.stroke();
```

Comment:
I was wondering if for each of your drawing functions it would not be safer to encapsulate the logic in a ctx.save() ... ctx.restore() ?

Reply:
keep save/restore in the loop, moved style methods above `beginPath()` and removed useless methods